internal/controller/serverset/serverset.go

```diff
@@ -46,9 +46,9 @@ const (
 
 const (
 	// indexLabel is the label used to identify the server set by index
-	indexLabel = "ionoscloud.com/serverset-%s-index"
+	indexLabel = "ionoscloud.com/%s-%s-index"
```

Comment:
shouldn't it be like this `"ionoscloud.com/%s-%d-index"`, because the second argument that will be expected is a integer. Please do the same for `versionLabel`.